packages/formdescriptioneditors/backend/sirius-components-collaborative-formdescriptioneditors/src/main/java/org/eclipse/sirius/components/collaborative/formdescriptioneditors/FormDescriptionEditorEventProcessor.java

```diff
@@ -129,6 +129,12 @@ public void refresh(ChangeDescription changeDescription) {
                 this.formDescriptionEditorContext.update(refreshedFormDescriptionEditor);
                 this.formDescriptionEditorEventFlux.formDescriptionEditorRefreshed(changeDescription.getInput(), refreshedFormDescriptionEditor);
             }
+        } else if (changeDescription.getKind().equals(ChangeKind.RELOAD_REPRESENTATION) && changeDescription.getSourceId().equals(this.formDescriptionEditorContext.getFormDescriptionEditor().getId())) {
+            Optional<FormDescriptionEditor> reloadedFormDescriptionEditor = this.representationSearchService.findById(this.editingContext, this.formDescriptionEditorContext.getFormDescriptionEditor().getId(), FormDescriptionEditor.class);
+            if (reloadedFormDescriptionEditor.isPresent()) {
+                this.formDescriptionEditorContext.update(reloadedFormDescriptionEditor.get());
+                this.formDescriptionEditorEventFlux.formDescriptionEditorRefreshed(changeDescription.getInput(), reloadedFormDescriptionEditor.get());
+            }
```

Comment:
I took upon myself to add tests last week to make this PR go through the continuous integration without issues and there something I don't understand here because I can't succeed in making this code work. The data structure persisted in the database is not the same as the one sent back to the frontend, the refresh is needed to compute the whole structure since it is not persisted. See these two PRs that you've reviewed before (#1484 and #1963) for additional details.

Without a new refresh and unless I've missed something, I think this will only reload an empty representation. I suspect that the same would be true for at least for the form representation which behave the same way, see `CreateFormEventHandler#handle` which does not store the real content of the form in the database. Have you ever seen this code work for forms or form description editors?

I'll push in another commit the tests that I've started to add, could you make the form description editor test succeed and contribute a XxxxLifecycleControllerTests for each representation kind (I've successfully done the Gantt one) to ensure that the mechanism works please.

Reply:
I'm on it.

Reply:
Should be fixed. I've added a small test for portals to increase the coverage. There's still a coverage issue with the Hierarchy representation, but it's not clear how to properly test it or of its even relevant.